src/pyuvdata/testing/warning_check.py

```diff
@@ -106,7 +106,11 @@ def __enter__(self):
             raise RuntimeError(f"Cannot enter {self!r} twice")
         _list = super().__enter__()
         # record=True means it's None.
-        assert _list is not None
+        if _list is None:  # pragma: no cover
+            raise RuntimeError(
+                "Something is wrong, _list should not be None. Please file an "
+                "issue in our GitHub issue log so that we can fix it."
```

Comment:
considering this is the warning checker, I cannot imagine many scenarios where this context would be entered without the user knowing. That said, if this error _somehow_ triggered on a test in a package like uvsim which uses this checker it could be ambiguous where to open an issue.

Should we mention something like the following? 

 - Something went wrong in pyuvdata's warning checker. Please file an issue in our GitHub issue log so that we can help: https://github.com/RadioAstronomySoftwareGroup/pyuvdata/issues. Developer info: _list is None

that's pretty verbose though 😕